### PR TITLE
elasticsearch:opensearch: add documentation for aws_service_name option

### PR DESCRIPTION
**Pull request by matthewfala**
# OpenSearch Serverless Support

Added the following option to support serverless:
```
AWS_Service_Name aoss
```

See the following PRs that are in review:
https://github.com/fluent/fluent-bit/pull/6448
https://github.com/fluent/fluent-bit/pull/6612

### CODE REVIEW

File: pipeline/outputs/opensearch.md [hunk 1]
```diff
@@ -21,10 +21,11 @@ The following instructions assumes that you have a fully operational OpenSearch
...
 | AWS\_External\_ID | External ID for the AWS IAM Role specified with `aws_role_arn`                                                                                                                                                                                                                                                                                                                                                                                               |  |
+| AWS\_Service\_Name | Service name to be used in AWS Sigv4 signature. For integration with Amazon OpenSearch Serverless, set to `aoss`                                                                                                                                                                                                                                                                                                                                                                                                | es |
```
Comment (PettitWesley) [inline pipeline/outputs/opensearch.md:24]: Can we add a serverless example?

Comment (lizsnyder) [reply pipeline/outputs/opensearch.md:24]: Nit: Can you add a period

Comment (matthewfala) [reply pipeline/outputs/opensearch.md]: Added an example and a period.

File: pipeline/outputs/opensearch.md [hunk 2]
```diff
@@ -157,3 +158,21 @@ Without this you will see errors like:
 ```text
 {"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"Action/metadata line [1] contains an unknown parameter [_type]"}],"type":"illegal_argument_exception","reason":"Action/metadata line [1] contains an unknown parameter [_type]"},"status":400}
 ```
+
+### Amazon OpenSearch Serverless <a id="#aws-opensearch-serverless"></a>
+Amazon OpenSearch Serverless an offering that eliminates your need to manage OpenSearch clusters. All existing Fluent Bit OpenSearch output plugin options work with OpenSearch Serverless. The only difference with serverless from a Fluent Bit POV is that you must specify the service name as `aoss` (Amazon OpenSearch Serverless) when you enable `AWS_Auth`:
```
Comment (lizsnyder) [inline pipeline/outputs/opensearch.md:163]: *is an offering

Comment (lizsnyder) [inline pipeline/outputs/opensearch.md:163]: For Fluent Bit, the only difference is that you must specify the service name...

File: pipeline/outputs/opensearch.md [hunk 3]
```diff
@@ -157,3 +158,21 @@ Without this you will see errors like:
...
+```
+AWS_Auth On
+AWS_Region <aws-region>
+AWS_Service_Name aoss
+```
+
+**Data Access Permissions**
+
+When sending logs to OpenSearch Serverless, your AWS IAM entity needs [OpenSearch Serverless Data Access permisions](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/serverless-data-access.html). Give your IAM entity the following Data Access permissions to your serverless collection:
```
Comment (lizsnyder) [inline pipeline/outputs/opensearch.md:172]: "data access" (not capitalized)

File: pipeline/outputs/opensearch.md (hunk 2 above)
Comment (matthewfala) [reply pipeline/outputs/opensearch.md]: updated

Comment (matthewfala) [reply pipeline/outputs/opensearch.md]: Updated

File: pipeline/outputs/opensearch.md (hunk 3 above)
Comment (matthewfala) [reply pipeline/outputs/opensearch.md]: updated

**Review by lizsnyder — APPROVED** [review]

File: pipeline/outputs/elasticsearch.md [hunk 4]
```diff
@@ -21,6 +21,7 @@ The **es** output plugin, allows to ingest your records into an [Elasticsearch](
 | AWS\_STS\_Endpoint | Specify the custom sts endpoint to be used with STS API for Amazon OpenSearch Service |  |
 | AWS\_Role\_ARN | AWS IAM Role to assume to put records to your Amazon cluster |  |
 | AWS\_External\_ID | External ID for the AWS IAM Role specified with `aws_role_arn` |  |
+| AWS\_Service\_Name | Service name to be used in AWS Sigv4 signature. For integration with Amazon OpenSearch Serverless, set to `aoss`. See the [FAQ](opensearch.md#faq) section on Amazon OpenSearch Serverless for more information. | es |
```
Comment (PettitWesley) [inline pipeline/outputs/elasticsearch.md:24]: the FAQ link doesn't work on this page

Comment (PettitWesley) [inline pipeline/outputs/elasticsearch.md:24]: I'm worried this relative FAQ won't work but I guess it does...

**Review by PettitWesley — APPROVED** [review]

File: pipeline/outputs/elasticsearch.md (hunk 4 above)
Comment (matthewfala) [reply pipeline/outputs/elasticsearch.md:24]: I think it does work. I tried it in vscode and the reference was found. In the diff, it takes me to `https://github.com/fluent/fluent-bit-docs/blob/aoss/pipeline/outputs/opensearch.md`

**Pull request closed without merging**